index.d.ts

```diff
@@ -0,0 +1,20 @@
+type Styles = string | TemplateStringsArray | string[] | Record<string, boolean>;
+
+declare function process(theme: Record<string, unknown>): (strings: Styles) => Record<string, unknown>;
+declare function configure(theme: Record<string, unknown>): Record<string, unknown>;
+declare function themed(theme?: Record<string, unknown>): (strings: Styles) => string;
```

Comment:
I'm no TS expert but doesn't calling themed return a `process`?

Reply:
Do you mean that themed should return a `process` or that it does?

Reply:
Unless I'm misinterpreting, calling `themed` returns a function that takes `Styles` as a param and returns a `string` for the class name.

```
export const themed = (theme = {}) => {
  const processWithTheme = process(configure(theme));
  return (strings, ...values) => css(processWithTheme(strings, values));
};
```

Reply:
Ahh yeh, my bad, I misread the signature. In that case.. this looks good!